x-pack/plugins/cloud/common/types.ts

```diff
@@ -5,4 +5,35 @@
  * 2.0.
  */
 
+import type { KibanaRequest } from '@kbn/core/server';
+
 export type OnBoardingDefaultSolution = 'es' | 'oblt' | 'security';
+
+/* FUTURE Engineer
```

Comment:
I am surprised by this hack that you had to do. I thought that for TS types circular dependency was not an issue. PluginA can import types from PluginB and vice versa... Isn't that possible?

Reply:
Unfortunately no, it's not possible, at least between plugins and/or packages. Each plugin is a TS project, and you can't cross-import between projects...